rgdc/rgdc/rgdc.py

```diff
@@ -206,19 +206,85 @@ def search(
         relates: Optional[str] = None,
         distance: Optional[Tuple[float, float]] = None,
         acquired: Optional[DATETIME_OR_STR_TUPLE] = None,
-        created: Optional[DATETIME_OR_STR_TUPLE] = None,
-        modified: Optional[DATETIME_OR_STR_TUPLE] = None,
-        datatype: Optional[SEARCH_DATATYPE_CHOICE] = None,
+        instrumentation: Optional[str] = None,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> List[Dict]:
+        """
+        Search for geospatial entries based on various criteria.
+
+        For Ranges (Tuples), an entry of `None` means that side of the range is unbounded.
+        E.g. a range of (2, None) is 2 or more, (None, 5) is at most 5, (2, 5) is between 2 and 5.
+
+        Args:
+            query: Either a WKT GeoJSON representation, a GeoJSON string, or a GeoJSON dict.
+            predicate: A named spatial predicate based on the DE-9IM. This spatial predicate will
+                be used to filter data such that predicate(a, b) where b is the queried geometry.
+            relates: Specify exactly how the queried geometry should relate to the data using a
+                DE-9IM string code.
+            distance: The min/max distance around the queried geometry in meters.
+            acquired: The min/max date and time (ISO 8601) when data was acquired.
+            instrumentation: The instrumentation used to acquire at least one of these data.
+            limit: The maximum number of results to return.
+            offset: The number of results to skip.
+
+        Returns:
+            A list of Spatial Entries.
+        """
+        # The dict that will be used to store params.
+        # Initialize with queries that won't be additionally processed.
+        params = {
+            'predicate': predicate,
+            'relates': relates,
+            'instrumentation': instrumentation,
+            'limit': limit,
+            'offset': offset,
+        }
+
+        if query:
+            if isinstance(query, str):
+                try:
+                    query = json.loads(query)
+                except JSONDecodeError:
+                    pass
+
+            if isinstance(query, dict):
+                # Allow failure on invalid format
+                query = wkt.dumps(query)
+
+            params['q'] = query
+
+        # Process range params
+
+        if distance and len(distance) == 2:
+            dmin, dmax = distance
+            params['distance_min'] = dmin
+            params['distance_max'] = dmax
+
+        # TODO: Determine if the before/after param order needs to be swapped?
+        if acquired and len(acquired) == 2:
+            amin, amax = acquired
+            params['acquired_before'] = datetime_to_str(amax)
+            params['acquired_after'] = datetime_to_str(amin)
+
+        return list(limit_offset_pager(self.session, 'geosearch', params=params))
+
+    def search_raster_stac(
```

Comment:
I'm trying to think about this in a similar way to the `get_raster` method, for which we combined the two methods into one and used the `stac` parameter to specify the difference. This is obviously more complex so I'm not sure it will apply in quite the same way, but there is a good deal of duplicated code between this and the regular search endpoint. @banesullivan Do you think it's worthwhile to try to bridge the gap between these two methods? I'm okay with keeping them completely separate, at least for now.

Reply:
I added a new `_spatial_search_params` to build the params for the SpatialEntry search which both methods can use to reduce duplicate code.

I do not want to combine these two methods because they are operating on different data types (even when we had the datatype param, we were only operating on SpatialEntry records).

Reply:
Done

Reply:
Did you push your changes? I don't see anything new